src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using CommunityToolkit.Maui.SourceGenerators.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Generators;
+
+[Generator]
+class TextColorToGenerator : IIncrementalGenerator
+{
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		// Get All Classes in User Library
+		var candidateClassesProvider = context.SyntaxProvider.CreateSyntaxProvider(
+			(syntaxNode, cancellationToken) => syntaxNode is ClassDeclarationSyntax,
+			(context, cancellationToken) => (ClassDeclarationSyntax)context.Node);
+
+		// Get Microsoft.Maui.Controls Assymbly Symbol
+		var mauiControlsAssemblySymbolProvider = context.CompilationProvider.Select(
+			(compilation, token) => compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == "Microsoft.Maui.Controls"));
+
+		var inputs = candidateClassesProvider.Collect()
+						.Combine(mauiControlsAssemblySymbolProvider)
+						.Select((combined, cancellationToken) => (CandidateClassesProvider: combined.Left, MauiControlsAssemblySymbolProvider: combined.Right));
+		context.RegisterSourceOutput(inputs, (context, collectedValues) =>
```

Comment:
we don't have access to the `Compilation` object here, can you double-check that? Maybe I missed something.